lib/matplotlib/backends/backend_cairo.py

```diff
@@ -217,7 +217,7 @@ def draw_markers(self, gc, marker_path, marker_trans, path, transform,
             self._fill_and_stroke(
                 ctx, rgbFace, gc.get_alpha(), gc.get_forced_alpha())
 
-    def draw_image(self, gc, x, y, im):
+    def draw_image(self, gc, x, y, im, bbox=None):
```

Comment:
What is supposed to happen if the bbox corner is not consistent with the (x, y)?  Should this really rather be a new method `draw_image_2(self, gc, bbox, im): ...`?  This will make it easier to check whether third-party backends implement that method, too.  (Or if we keep the same name, we may need signature-sniffing tricks to figure out signature to use... or something like option_scale_image.)

Reply:
Didn't think of third-party backends. I guess I should implement it as a separate method.

Reply:
Actually, to prevent having to do some weird deprecation stuff to `draw_image`, I think the better solution is to just add optional `width` and `height` arguments instead, and yeah....a `hasattr` trick to check what signature to use would work....even though it feels icky to write. Like `option_exact_image_bbox`.